src/Header.tsx

```diff
@@ -1,10 +1,15 @@
 import styled, {css} from 'styled-components'
 import PropTypes from 'prop-types'
 import theme from './theme'
-import {get, COMMON, TYPOGRAPHY, BORDER} from './constants'
-import sx from './sx'
+import {BORDER, COMMON, get, SystemCommonProps, SystemTypographyProps, SystemBorderProps, TYPOGRAPHY} from './constants'
+import {ComponentProps} from './utils/types'
+import sx, {SxProp} from './sx'
 
-const Header = styled.div`
+type StyledHeaderItemProps = SystemCommonProps & SxProp & {full?: boolean}
+type StyledHeaderProps = SystemCommonProps & SxProp
```

Comment:
Header also accepts border props:

```suggestion
type StyledHeaderProps = SystemBorderProps & SystemCommonProps & SxProp
```